src/libstd/collections/hash/set.rs

```diff
@@ -337,7 +337,7 @@ impl<T, S> HashSet<T, S>
     /// assert_eq!(diff, [4].iter().collect());
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn difference<'a>(&'a self, other: &'a HashSet<T, S>) -> Difference<'a, T, S> {
+    pub fn difference<'a, U>(&'a self, other: &'a HashSet<T, U>) -> Difference<'a, T, U> {
```

Comment:
Should we uniformly require the BuildHasher bound here? Not sure if it would matter but it might be safer to add?

Reply:
Yeah, I find it hard to reason about inference issues in cases like this. I agree that it probably is a good idea to make explicit.